PsfLauncher/main.cpp

```diff
@@ -106,16 +147,46 @@ int launcher_main(PCWSTR args, int cmdShow) noexcept try
             return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
         }
 
-        error = RunScript(*startScriptInformation, packageRoot, dirStr, cmdShow);
+        PowershellScript startingScript(*startScriptInformation, currentDirectory);
+        bool canScriptRun = false;
+        error = CheckIfShouldRun(startingScript.GetRunOnce(), canScriptRun);
 
         if (error.IsThereAnError())
         {
             ::PSFReportError(error.Print().c_str());
             return error.GetErrorNumber();
         }
+
+        if (canScriptRun)

```

Comment:
Once all the script logic is in the PowershellScript object, this can be cleaned up a lot.